Series adds real-time capability to mpd threads

Upstream: n/a (local customization)

Signed-off-by: Andreas Ziegler <15275159+aeolio@users.noreply.github.com>
---
--- a/src/Main.cxx
+++ b/src/Main.cxx
@@ -50,6 +50,10 @@
 #include "io/linux/ProcStatus.hxx"
 #endif
 
+#ifdef ENABLE_RTOPT
+#include "thread/RealtimeOption.hxx"
+#endif
+
 #ifdef ENABLE_DAEMON
 #include "unix/Daemon.hxx"
 #endif
@@ -335,6 +339,11 @@
 
 	log_init(raw_config, options.verbose, options.log_stderr);
 
+#ifdef ENABLE_RTOPT
+	RealtimeOption::Initialize(raw_config);
+	RealtimeOption::ChangeThreadPriority("main");
+#endif
+
 	Instance instance;
 	global_instance = &instance;
 
@@ -378,6 +387,10 @@
 	spl_global_init(raw_config);
 #ifdef ENABLE_ARCHIVE
 	const ScopeArchivePluginsInit archive_plugins_init{raw_config};
+#endif
+
+#ifdef ENABLE_RTOPT
+	RealtimeOption::LockMemory();
 #endif
 
 	pcm_convert_global_init(raw_config);
